### PR TITLE
Feature/13 openapi

### LINKED ISSUE #13: OpenAPI integration
## Background
Create a blueprint of the OpenAPI that would help us generate useful things - client libraries, documentation, Swagger doc site etc.

If unable to directly generate backend API (assuming such a generator missing for our combination of Scala+Spring), attempt to use the generator for close combination and enhance our hand-written API manually -- by being inspired with the generator output at least.

### PR DESCRIPTION
**Pull request by dk1844**
Initial OpenAPI integration.

## OpenAPI 
 - OpenAPI v3.0 is available
   - Offers API doc at http://localhost:8080/v3/api-docs & http://localhost:8080/v3/api-docs.yaml
 - Implemented using `org.springdoc:springdoc-openapi-ui` (replacement of the late `springfox`).

## Current user credentials:
`TestUser` / `password123`

## Swagger 
Swagger is present, can be loaded at http://localhost:8080/swagger-ui/index.html#
image

Closes #13

### CODE REVIEW

File: service/src/main/scala/za/co/absa/logingw/rest/Application.scala [hunk 1]
```diff
@@ -19,12 +19,31 @@ package za.co.absa.logingw.rest
...
 import org.springframework.boot.web.servlet.support.SpringBootServletInitializer
 import org.springframework.context.annotation._
 
+@OpenAPIDefinition(
+  externalDocs = new ExternalDocumentation(description = "GitHub",
+    url = "https://github.com/AbsaOSS/login-gateway"
+  ),
+  info = new Info(
+    title = "Login Gateway API",
+    version = "0.1",
```
Comment (Zejnilovic) [inline service/src/main/scala/za/co/absa/logingw/rest/Application.scala:38]: Can these be filled from a config/sbt files? 
I presume they can't but I have to ask.

Comment (dk1844) [reply service/src/main/scala/za/co/absa/logingw/rest/Application.scala:38]: I believe they can be, similar to what we do in other project (Enceladus does this somehow). To be part of #5.

**Review by TheLydonKing — COMMENTED** [review]
Changes Make Sense to me.

May require a 2nd pair of eyes, however, as I am still relatively new to the language.

**Review by miroslavpojer — APPROVED** [review]
- [x] pull
- [x] code review
- [x] built
- [x] test ran
- [ ] manually tested in local tomcat

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala [hunk 2]
```diff
@@ -26,12 +26,12 @@ import org.springframework.test.context.TestContextManager
 class TestServiceTest extends AnyFlatSpec with Matchers {
```
Comment (jakipatryk) [inline service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala:26]: The class/file didn't get renamed.

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala [hunk 3]
```diff
@@ -26,12 +26,12 @@ import org.springframework.test.context.TestContextManager
...
 
   @Autowired
-  private var testService: TestService = _
+  private var testService: InfoService = _
```
Comment (jakipatryk) [inline service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala:29]:
```suggestion
  private var infoService: InfoService = _
```

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala [hunk 4]
```diff
@@ -26,12 +26,12 @@ import org.springframework.test.context.TestContextManager
...
 
   // Makes the above autowired work
   new TestContextManager(this.getClass).prepareTestInstance(this)
 
   "TestService" should "give expected test message" in {
```
Comment (jakipatryk) [inline service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala:34]:
```suggestion
  "InfoService" should "give expected test message" in {
```

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala [hunk 5]
```diff
@@ -26,12 +26,12 @@ import org.springframework.test.context.TestContextManager
...
-    testService.getTestMessage shouldEqual "Test service is OK"
+    testService.getInfoMessage shouldEqual "Basic info message to be here."
```
Comment (jakipatryk) [inline service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala:35]:
```suggestion
    infoService.getInfoMessage shouldEqual "Basic info message to be here."
```

**Review by jakipatryk — COMMENTED** [review]
LGTM (checked code/annotations, pulled, built, ran locally, checked the UI and played with it a bit, downloaded the YAML file). After this little renaming is done I'll approve, cool feature.

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala (hunk 5 above)
Comment (dk1844) [reply service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala]: Thanks for noticing, fixed.

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala (hunk 4 above)
Comment (dk1844) [reply service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala]: Thanks for noticing, fixed.

File: service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala (hunk 2 above)
Comment (dk1844) [reply service/src/test/scala/za/co/absa/logingw/rest/service/TestServiceTest.scala]: Thanks, fixed.

**Comment by Zejnilovic** [comment]
Did you look into the Spring actuator and its possible addition to the Swagger Docs?

**Review by jakipatryk — APPROVED** [review]

**Comment by dk1844** [comment]
> Did you look into the Spring actuator and its possible addition to the Swagger Docs?

Yes, this there already is an issue for this - https://github.com/AbsaOSS/login-gateway/issues/6

**Pull request merged**